src/System.Collections.Immutable/src/Properties/SecurityTransparent.cs

```diff
@@ -0,0 +1,6 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Security;
+
+[assembly: SecurityTransparent]
```

Comment:
I'm not entirely sure what the implications of this are so I cannot say definitively. Unless someone does some proper test validation and understands the scope I don't think I'm comfortable with blindly doing this.